state/allwatcher.go

```diff
@@ -1336,30 +1328,34 @@ type backingEntityDoc interface {
 
 func newAllWatcherStateBacking(st *State, params WatchParams) Backing {
 	collectionNames := []string{
+		// The ordering here matters. We want to load machines, then
```

Comment:
Is there a test you could write (or one that already exists) to confirm this ordering?

Reply:
The existing tests would have started failing due to the change in processing within a single unit if the ordering wasn't working.

The new code to determine the opened ports for a unit require the principal to have been processed first, otherwise it would raise errors.